cirq/work/observable_measurement_test.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2020 The Cirq developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pytest
+
+import cirq
+import cirq.work as cw
+from cirq.work.observable_measurement import (
+    _with_parameterized_layers,
+    _get_params_for_setting,
+    _pad_setting,
+    _subdivide_meas_specs,
+)
+
+
+def test_with_parameterized_layers():
+    qs = cirq.LineQubit.range(3)
+    circuit = cirq.Circuit(
```

Comment:
I'd assert the actual content of the parameterized layers as well...this test can be passed with just adding a couple of empty moments or moments containing identity gates.

Reply:
added gate-isinstance checks and checked the sympy symbol names in the parameterized layers here and for the needs_init_layer=True case